### PR TITLE
`schemadiff`: better nil check validation

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Quick followup to https://github.com/vitessio/vitess/pull/15519, when further testing `annotations`, we get `nil` reference errors. This is due to the ever unexpected `golang` behavior of "interface is not nil but actual value is nil".

This PR does better handling of `nil` interface values, and adds tests to validate.

## Related Issue(s)

https://github.com/vitessio/vitess/pull/15519

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/schemadiff/diff.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func DiffSchemas(env *Environment, schema1 *Schema, schema2 *Schema, hints *Diff
 
 // EntityDiffByStatement is a helper function that returns a simplified and incomplete EntityDiff based on the given SQL statement.
 // It is useful for testing purposes as a quick mean to wrap a statement with a diff.
-func EntityDiffByStatement(statement sqlparser.Statement) EntityDiff {
+func EntityDiffByStatement(env *Environment, statement sqlparser.Statement) EntityDiff {
```
Comment (rohit-nayak-ps) [inline go/vt/schemadiff/diff.go:180]: `env` doesn't appear to be actually used in this function?

**Review by rohit-nayak-ps — APPROVED** [review]

File: go/vt/schemadiff/diff.go (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/schemadiff/diff.go]: You're right. It was due to an earlier iteration. Now removed.

File: go/vt/schemadiff/annotations.go [hunk 2]
```diff
@@ -214,9 +214,16 @@ func unifiedAnnotated(from *TextualAnnotations, to *TextualAnnotations) *Textual
 // annotatedDiff returns the annotated representations of the from and to entities, and their unified representation.
 func annotatedDiff(diff EntityDiff, entityAnnotations *TextualAnnotations) (from *TextualAnnotations, to *TextualAnnotations, unified *TextualAnnotations) {
 	fromEntity, toEntity := diff.Entities()
+	// Handle the infamous golang interface is not-nil but underlying object is:
```
Comment (mattlord) [inline go/vt/schemadiff/annotations.go:217]: TiL that an interface holding a nil value is not nil

**Review by mattlord — APPROVED** [review]

File: go/vt/schemadiff/annotations.go (hunk 2 above)
Comment (shlomi-noach) [reply go/vt/schemadiff/annotations.go:217]: You're in for some fun times!

**Pull request merged**